oauth2_http/java/com/google/auth/oauth2/IdentityPoolCredentials.java

```diff
@@ -88,76 +103,17 @@ public AccessToken refreshAccessToken() throws IOException {
 
   @Override
   public String retrieveSubjectToken() throws IOException {
-    if (identityPoolCredentialSource.credentialSourceType
-        == IdentityPoolCredentialSource.IdentityPoolCredentialSourceType.FILE) {
-      return retrieveSubjectTokenFromCredentialFile();
-    }
-    return getSubjectTokenFromMetadataServer();
+    return this.subjectTokenProvider.getSubjectToken();
```

Comment:
Here's an example of us cheating in our own implementation, offering `Retryable` `IOExceptions` for our own implementations but losing our customers RTEs into a generic `IOException`.

Reply:
I believe this is resolved by the refactor to have the supplier/provider be public to users but feel free to reopen if I'm not understanding correctly